src/web_resources/HangarXPLOR.ProcessItem.js

```diff
@@ -9,136 +9,225 @@ HangarXPLOR._ltiCount      = HangarXPLOR._ltiCount || 0;
+    }
+    var comboShip = $(this)
+    if (debug) {
+      console.log('CleanComboShips comboShip BEFORE', comboShip);
+    }
+    var comboShipName = comboShip.text();
+    if (debug) {
+      console.log('CleanComboShips comboShipName BEFORE', comboShipName);
+    }
+    comboShipName = HangarXPLOR.CleanComboShipName(comboShipName);
+    if (debug) {
+      console.log('CleanComboShips comboShipName AFTER', comboShipName);
+    }
+    comboShip[0].textContent = comboShipName;
+    if (debug) {
+      console.log('CleanComboShips comboShip AFTER', comboShip);
+    }
+  })
+}
+
+HangarXPLOR.CleanComboShipName = function(comboShipName)
+{
+  comboShipName = comboShipName.replace('Tumbril Cyclone-TR', 'Tumbril Cyclone TR');
+  return comboShipName;
+}
+
 // Apply a pre-defined filter to a list of items
 HangarXPLOR.ProcessItem = function()
 {
-  // Pre-cache raw markup
-  if (HangarXPLOR._fromCache != true) HangarXPLOR._raw.push(this.outerHTML);
-
-  // Preprocess Tumbril Cyclone
-  $('.without-images .title:contains(Tumbril Cyclone)', this).each(function() {
-    var $tumbril = $(this);
-    var name = $tumbril.text().trim();
-    if (name == "Tumbril Cyclone") name = "Tumbril Cyclone-RN";
-    
-    if (name != "Tumbril Cyclone-AA" &&
-        name != "Tumbril Cyclone-TR" &&
-        name != "Tumbril Cyclone-RN" &&
-        name != "Tumbril Cyclone-RC") return;
-        
-    var $block = $tumbril.closest('.content-block1');
-    var $images = $('.with-images', $block);
-    if ($images.length == 0) {
-      $block.prepend($('<div />').addClass('also-contains').text("Also Contains"));
-      $block.prepend($images = $('<div />').addClass('with-images'));
-    }
-    
-    var cyclone = {
-      "Tumbril Cyclone-RN": "/media/ao2p3pw2e7k94r/subscribers_vault_thumbnail/Tumbril-Buggy-Piece-01-Showroom-V009.jpg",
-      "Tumbril Cyclone-TR": "/media/cmq3rwpo5ghpvr/subscribers_vault_thumbnail/Tumbril-Buggy-Piece-04-Desert-V012.jpg",
-      "Tumbril Cyclone-RC": "/media/w3vw5498xb25mr/subscribers_vault_thumbnail/Tumbril-Buggy-Piece-05-Rocky-Beach-Sport-Fin.jpg",
-      "Tumbril Cyclone-AA": "/media/n6535dpiwv2pgr/subscribers_vault_thumbnail/Tumbril-Buggy-Piece-06-Lagoon-V011.jpg",
-    };
-    
-    var $item = $('<div />').addClass('item').append(
-      $('<div />').addClass('image').css({ 'background-image': 'url("' + cyclone[name] + '")' }),
-      $('<div />').addClass('text').append(
-        $('<div />').addClass('title').text(name),
-        $('<div />').addClass('kind').text("Ship"),
-        $('<div />').addClass('liner').append("Tumbril (", $('<span />').text("TMBL"), ")")
-      )
-    );
-    $images.prepend($item);
-    $tumbril.parent().remove();
+  // Preprocessing Begin
+  // Tumbril Cyclone
+  $('.without-images .title:contains(Tumbril Cyclone)', this).each(function(index, value) {
+    //console.log('ProcessItem PreProcessTumbrilCyclone index=' + index + ', value=', value);
+    HangarXPLOR.PreProcessTumbrilCyclone($(this));
```

Comment:
Not a bad idea to convert different chunks of the cleaning code into discrete functions